packages/test/test-utils/src/opProcessingController.ts

```diff
@@ -75,13 +80,44 @@ class DeltaManagerMonitor extends DeltaManagerToggle {
     private pendingCount: number = 0;
     private clientId: string | undefined;
     private firstClientSequenceNumber: number = -1;
-    private _latestSequenceNumber: number;
+    private lastOutbound: IDocumentMessage | undefined;
+    private readonly lastInboundPerClient = new Map<string, ISequencedDocumentMessage>();
+    private pendingWriteConnection = false;
+
+    public expectingInboundFrom(outbound: DeltaManagerMonitor): boolean {
+        // there should be no outstanding work for disposed delta managers
+        if (this.deltaManager.disposed || outbound.deltaManager.disposed) {
+            return false;
+        }
+        // if there is no last outbound, we are not waiting for anything
+        if (outbound.lastOutbound === undefined
+            || outbound.clientId === undefined) {
+            return false;
+        }
+        // if out inbound is paused we are not expecting to receive anything more
+        if (this.inboundPaused) {
+            return false;
+        }
+
+        // if outbound is ourself, return if we having pending work
+        if (this === outbound) {
+            return this.hasPendingWork();
+        }
+
+        // check if we are waiting to see a message from outbound
+        const lastInboundForOutbound = this.lastInboundPerClient.get(outbound.clientId);
+        if (lastInboundForOutbound !== undefined) {
+            return outbound.lastOutbound.clientSequenceNumber > lastInboundForOutbound.clientSequenceNumber;
```

Comment:
Note that this check can lie due to noop coalescing on the server. I.e. it can result in false positive.

Reply:
i'm not sure how that would happen. no-op are excluded from lastOutbound and server NoOp should have a null client id so won't be tracked in lastInboundPerClient.